src/prefetcher.cpp

```diff
@@ -26,36 +26,130 @@
 
 #include "prefetcher.h"
 #include "bithacks.h"
+#include "hash.h"
+#include <limits>
+#include "g_std/g_list.h"
+#include <unordered_set>
 
 //#define DBG(args...) info(args)
 #define DBG(args...)
 
+void StreamPrefetcher::postInit() {
```

Comment:
why is this copied from basecache? This class should just inherit from basecache.